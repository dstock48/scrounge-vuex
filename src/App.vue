<template>
  <main id="app">
    <Header />
    <IngredientEntry />
    <IngredientsContainer />
    <RecipesContainer />
  </main>
</template>

<script>
import Header from "./components/Header";
import IngredientEntry from "./components/IngredientEntry";
import RecipesContainer from "./components/RecipesContainer";
import IngredientsContainer from "./components/IngredientsContainer";

export default {
  name: "App",
  components: {
    Header,
    IngredientEntry,
    RecipesContainer,
    IngredientsContainer
  }
};
</script>

<style scoped lang="scss">
@import url('https://fonts.googleapis.com/css?family=Roboto:100,300,400,500,700,900&display=swap');
main {
  font-family: "Roboto", sans-serif;
}
</style>
