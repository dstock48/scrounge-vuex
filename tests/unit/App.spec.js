import { shallowMount, createLocalVue } from "@vue/test-utils";
import App from "../../src/App.vue";
import Vuex from "vuex";

const localVue = createLocalVue();

localVue.use(Vuex);

describe("App.vue", () => {
  let state;
  let actions;
  let store;

  beforeEach(() => {
    state = {
      ingredients: ["mockPeas", "mockCarrots"],
      selectedRecipe: { title: "mock" },
      recipeIsOpen: true
    }
    actions = {
      toggleOpenRecipe: jest.fn(),
    };
    store = new Vuex.Store({
      state,
      actions
    });
  });

  it("matches the snapshot", () => {
    const wrapper = shallowMount(App);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("dispatches toggleOpenRecipe when recipe card is clicked", () => {
    const wrapper = shallowMount(App, { store, localVue });
    wrapper.find(".toggle-icon-wrapper").trigger("click");
    expect(actions.toggleOpenRecipe).toHaveBeenCalled();
  });
});
