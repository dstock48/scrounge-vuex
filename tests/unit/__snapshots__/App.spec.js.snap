// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue matches the snapshot 1`] = `
<main
  id="app"
>
  <header-stub />
   
  <div
    class="sticky-container"
  >
    <ingrediententry-stub />
     
    <p
      class="default-message"
    >
      
      We'll show you recipes that match the ingredients you have.
    
    </p>
     
    <!---->
  </div>
   
  <div
    class="recipe-section-wrapper"
  >
    <div
      class="home-image"
    >
      <h4>
        Turn the food you have left into the meal you'll make again.
      </h4>
    </div>
     
    <!---->
  </div>
</main>
`;
